#523 Create regression test for /admin/project/list [2-4h]

raidenovski opened the pull request:
In order to be able to run selenium tests on newer versions of firefox without problems, I have upgraded the selenium dependency to 3.0.1.
In order to use it, we need gecko driver (new web driver for Firefox). 
I used **gecko driver 0.14**, anything higher causes bugs. 

**Should the build fail** because of tests, we need to add this gecko driver to our server. 
In order to do that: 
Download driver from github: 
`wget https://github.com/mozilla/geckodriver/releases/download/v0.14.0/geckodriver-v0.14.0-linux64.tar.gz` 

Extract: `tar -xvzf geckodriver*` 

Make it executable: `chmod +x geckodriver`

 Add it to our bin: `sudo mv geckodriver /usr/local/bin/`

jo-elimuai commented on .gitignore:
```diff
@@ -6,3 +6,7 @@ pom.xml.next
+.classpath
+.project
+.settings
+.gitignore
```
@raidenovski The `.gitignore` file itself probably shouldn't be ignored?

raidenovski replied:
Will remove it from `.gitignore`

jo-elimuai commented on pom.xml:
```diff
@@ -525,13 +525,11 @@
+            <version>3.0.1</version>
         </dependency>
-        
         <!-- Needed by Selenium to avoid "java.lang.NoClassDefFoundError: org/apache/http/conn/HttpClientConnectionManager" -->
```
@raidenovski Can this be removed now that the Selenium version has been upgraded, or does the `NoClassDefFoundError` error still appear during compilation?

raidenovski replied:
@jo-elimuai Haven't tried it yet but will do so

jo-elimuai commented on src/test/java/selenium/DomainHelper.java:
```diff
@@ -3,8 +3,8 @@
     public static String getBaseUrl() {
-      //return "http://localhost:8080/webapp";
-        return "http://test.elimu.ai";
+      return "http://localhost:8080/webapp";
```
@raidenovski This URL will not work on the test server.

raidenovski replied:
@jo-elimuai My mistake, will change

jo-elimuai commented:
> I used gecko driver 0.14, anything higher causes bugs.

@raidenovski Which Firefox version did you use?

raidenovski commented:
> @raidenovski Which Firefox version did you use?

@jo-elimuai Firefox 55.0, Linux mint 18

raidenovski commented:
@jo-elimuai Removing the `org.apache.httpcomponents` dependency causes an error on [MailChimpApiHelper](https://github.com/elimu-ai/webapp/blob/master/src/main/java/ai/elimu/util/MailChimpApiHelper.java#L87) because it is in use.
Changed the .gitignore and switched the `DomainHelper` to return the test server url
